src/lib/y2network/network_manager/connection_config_writers/base.rb

```diff
@@ -40,23 +40,36 @@ def initialize(file)
         # Writes connection information to the interface configuration file
         #
         # @param conn [Y2Network::ConnectionConfig::Base] Connection to take settings from
-        def write(conn)
+        # @param routes [<Array<Route>]
+        def write(conn, routes = [])
           file.connection["id"] = conn.name
           file.connection["autoconnect"] = "false" if ["manual", "off"].include? conn.startmode.name
           file.connection["permissions"] = nil
           file.connection["interface-name"] = conn.interface
           file.connection["zone"] = conn.firewall_zone unless ["", nil].include? conn.firewall_zone
-          conn.bootproto.dhcp? ? configure_dhcp(conn) : add_ips(conn)
+          conn.bootproto.dhcp? ? configure_dhcp(conn) : configure_ips(conn)
+          configure_routes(conn, routes)
           update_file(conn)
         end
 
       private
 
+        def configure_routes(conn, routes)
+          routes.select { |r| (r.interface&.name == conn.name) && r.is_default? }.each do |route|
```

Comment:
I would pass only the relevant routes (from this code, I infer that we are passing all the routes).

Reply:
Originally I passed only the routes for the device, but then realized that we probably will need also the global routes so decided to pass all the routes by now, but can move it back to the config writer.